pyat/test/test_basic_elements.py

```diff
@@ -139,6 +140,162 @@ def test_multipole():
+        warnings.simplefilter("error")
+        if element_type is elements.ThinMultipole:
+            elem = element_type("ThinMulti", [], [0.0, 0.0, 0.0], k=1.0, h=1.0)
+        else:
+            elem = element_type("Multi", 1.0, [], [0.0, 0.0, 0.0], k=1.0, h=1.0)
+    assert elem.PolynomB[1] == 1.0
+    assert elem.PolynomB[2] == 1.0
+    # No warnings or changes when PolynomB[1] and K are the same but not default values
+    # and PolynomB[2] and H are the same but not default values
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        if element_type is elements.ThinMultipole:
+            elem = element_type("ThinMulti", [], [0.0, 1.0, 2.0], k=1.0, h=2.0)
+        else:
+            elem = element_type("Multi", 1.0, [], [0.0, 1.0, 2.0], k=1.0, h=2.0)
+    assert elem.PolynomB[1] == 1.0
+    assert elem.PolynomB[2] == 2.0
+    # Change and warning when PolynomB[1] is not default value and replaces K
+    with pytest.warns(lattice.AtWarning):
+        if element_type is elements.ThinMultipole:
+            elem = element_type("ThinMulti", [], [0.0, 1.0, 0.0], k=2.0)
+        else:
+            elem = element_type("Multi", 1.0, [], [0.0, 1.0, 0.0], k=2.0)
+    assert elem.K == 1.0
+    assert elem.PolynomB[1] == 1.0
+    # Change and warning when PolynomB[2] is not default value and replaces H
+    with pytest.warns(lattice.AtWarning):
+        if element_type is elements.ThinMultipole:
+            elem = element_type("ThinMulti", [], [0.0, 0.0, 1.0], h=2.0)
+        else:
+            elem = element_type("Multi", 1.0, [], [0.0, 0.0, 1.0], h=2.0)
+    assert elem.H == 1.0
+    assert elem.PolynomB[2] == 1.0
+
+
+@pytest.mark.parametrize("element_type", [elements.Quadrupole, elements.Dipole])
+def test_quadrupolar_strength_prioritisation(element_type):
+    # No warnings or changes when K isn't specified
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        if element_type is elements.Quadrupole:
+            elem = element_type("Quad", 1.0, PolynomB=[0.0, 1.0])
+        else:
+            elem = element_type("Dipole", 1.0, 0.0, PolynomB=[0.0, 1.0])
+    assert elem.PolynomB[1] == 1.0
+    # No warnings or changes when K is the default value (0.0)
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        if element_type is elements.Quadrupole:
+            elem = element_type("Quad", 1.0, k=0.0, PolynomB=[0.0, 1.0])
+        else:
+            elem = element_type("Dipole", 1.0, 0.0, k=0.0, PolynomB=[0.0, 1.0])
+    assert elem.PolynomB[1] == 1.0
+    # No warnings or changes when K is None
+    with warnings.catch_warnings():
+        warnings.simplefilter("error")
+        if element_type is elements.Quadrupole:
+            elem = element_type("Quad", 1.0, k=None, PolynomB=[0.0, 1.0])
+        else:
+            elem = element_type("Dipole", 1.0, 0.0, k=None, PolynomB=[0.0, 1.0])
+    assert elem.PolynomB[1] == 1.0
```

Comment:
`None` is not an allowed value for the strength. If you don't specify PolynomB you get a nan in the element: not very useful.

You can remove this test, and feel free to change the type hint in the Quadrupole definition and suppress the None